mixer/pkg/server/server.go

```diff
@@ -162,10 +163,29 @@ func newServer(a *Args, p *patchTable) (server *Server, err error) {
 
 	if network == "unix" {
 		// remove Unix socket before use.
-		if err = p.remove(address); err != nil && !os.IsNotExist(err) {
+		if err = p.remove(address); err != nil && !os.IsPermission(err) && !os.IsNotExist(err) {
 			// Anything other than "file not found" is an error.
 			return nil, fmt.Errorf("unable to remove unix://%s: %v", address, err)
 		}
+		// if no exist, try to run command with sudo to give permission for folder
+		if os.IsNotExist(err) {
+			if strings.Contains(address, "/") {
+				parentPath := address[0:strings.LastIndex(address, "/")]
+				if _, err := os.Stat(parentPath); err != nil && os.IsPermission(err) {
+					cmd := exec.Command("/bin/sh", "-c", "sudo chmod -R 666 "+parentPath)
```

Comment:
running sudo feels a bit off to me.. what's the point of dropping root if anyone can just run sudo anyways? Is this a common pattern to run sudo within a server?

Reply:
Could you elaborate more about `dropping root if anyone can just run sudo anyways`? I think this should be a common pattern to run sudo.

Reply:
I am not experienced in this topic, but my thought was that running as non-root is to prevent attackers from accessing privileged things. If they can just run `sudo` what do we gain?

@fpesce can you weigh in here?

Reply:
@clyang82 In the Helm chart UDS is hardcoded right? c.f. https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/deployment.yaml#L53-L54
From a security perspective, I'm not a huge fan of passing an exec.Command parameter with content that can be submit by user (+parentPath) without any kind of sanitization.

Reply:
Changing the permissions of the UDS can probably be in the Dockerfile.

Reply:
The reason for this shell command was that Mixer did not clean up UDS file on crash. The files needs to be removed before creating a new server socket.

Reply:
Ok. Then, if this fix is absolutely needed, it should drop exec.Command and use os.Chmod() and os.Remove()

Reply:
non root user does not have permission to use os.Chmod() and os.Remove(). That is why introduce exec.Command here. Any suggestions? Thanks.

Reply:
sorry for misleading here. I just double checked, the fold permission is 777 if it is created from emptyDir. and the istio-proxy user has permission to run os.Remove to remove /sock/mixer.socket file.
```
$ ls -l /sock/
total 0
srwxr-xr-x 1 istio-proxy istio-proxy 0 Oct 11 03:13 mixer.socket
$ ls -l /
total 4
drwxr-xr-x   1 root root  125 Sep 20 20:10 bin
drwxr-xr-x   2 root root    6 Apr 24  2018 boot
drwxr-xr-x   5 root root  360 Oct 11 03:13 dev
drwxr-xr-x   1 root root   19 Oct 11 03:13 etc
drwxr-xr-x   1 root root   25 Sep 20 20:11 home
drwxr-xr-x   1 root root   45 May 23  2017 lib
drwxr-xr-x   2 root root   34 Sep 12 20:53 lib64
drwxr-xr-x   2 root root    6 Sep 12 20:53 media
drwxr-xr-x   2 root root    6 Sep 12 20:53 mnt
drwxr-xr-x   2 root root    6 Sep 12 20:53 opt
dr-xr-xr-x 426 root root    0 Oct 11 03:13 proc
drwx------   2 root root   37 Sep 12 20:54 root
drwxr-xr-x   1 root root   21 Oct 11 03:13 run
drwxr-xr-x   1 root root 4096 Sep 20 20:10 sbin
drwxrwxrwx   2 root root   26 Oct 11 03:13 sock
drwxr-xr-x   2 root root    6 Sep 12 20:53 srv
dr-xr-xr-x  13 root root    0 Sep 23 09:31 sys
drwxrwxrwt   1 root root    6 Sep 20 20:10 tmp
drwxr-xr-x   1 root root   19 Sep 12 20:53 usr
drwxr-xr-x   1 root root   17 Sep 12 20:54 var
```